brave-cxf3/src/main/java/com/github/kristofa/brave/cxf3/BraveCxfExecutorService.java

```diff
@@ -0,0 +1,157 @@
+package com.github.kristofa.brave.cxf3;
+
+import com.github.kristofa.brave.Brave;
+import com.github.kristofa.brave.ServerSpan;
+import com.github.kristofa.brave.ServerSpanThreadBinder;
+import org.apache.cxf.message.Message;
+import org.apache.cxf.phase.PhaseInterceptorChain;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.*;
+
+/**
+ * Wraps around {@link ExecutorService} and passes {@link ServerSpan}
+ * from cxf {@link org.apache.cxf.message.Exchange} context (if exists)
+ * to current {@link ServerSpanThreadBinder} during execution.
+ */
+public class BraveCxfExecutorService implements ExecutorService, Closeable {
+
+  private final ExecutorService delegate;
```

Comment:
You actually would better do that:
```
public void foo(@Context TracerContext context, @Suspended AsyncResponse response) {   
        executor.execute(context.wrap(() -> {
          response.resume("foo");
        });
      });
    }
```

In this case the executor is application specific, while `tracerContext` wraps callable / runnable into scope. Plus, it will work with any executor :-) What do you think?